File: CSV2HTML_by_Java/csv2html/Reader.java
package csv2html;

import java.io.File;
import java.util.List;

/**
 * リーダ：情報を記したCSVファイルを読み込んでテーブルに仕立て上げる。
 */
public class Reader extends IO
{
	/**
	 * リーダのコンストラクタ。
	 * @param aTable テーブル
	 */
	public Reader(Table aTable)
	{
		super(aTable);

		return;
	}

	/**
	 * ダウンロードしたCSVファイルを読み込む。
	 */
	public void perform()
	{
		String csvUrl = super.attributes().csvUrl();
		List<String> separateListUrl = IO.splitString(csvUrl,"/");
		File csvFile = new File(super.attributes().baseDirectory(),separateListUrl.get(separateListUrl.size() - 1));
		List<String> csvListText = IO.readTextFromFile(csvFile);

		/*
		 * ここは正規表現で書ける可能性があるかも by北澤 12月9日
		 */
		String name = csvListText.get(0);
		super.attributes().names(IO.splitString(name,","));
		//System.out.println(IO.splitString(name,","));

		/*
		 * for文で書いてるけどもっとスマートなやり方があるかも
		 */
		for(int i=1;i<csvListText.size();i++){
			List<String> splitText = IO.splitString(csvListText.get(i),",");
			Tuple aTuple = new Tuple(super.attributes(), splitText);
			super.table().add(aTuple);
		}


		return;
	}
}

File: CSV2HTML_by_Python/csv2html/reader.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""CSV2HTML：総理大臣と徳川幕府の情報「CSV」からWebページ「HTML」を生成。"""
__author__ = 'KiTAZAWA SHOTA'
__version__ = '1.1.0'
__date__ = '2024/12/14 (Created: 2024/12/14)'

import os

from csv2html.io import IO
from csv2html.tuple import Tuple

class Reader(IO):
	"""リーダ：情報を記したCSVファイルを読み込んでテーブルに仕立て上げる。"""

	def __init__(self, input_table):
		"""リーダのコンストラクタ。"""

		super().__init__(input_table)
		(lambda x: x)(input_table) # NOP

	def perform(self):
		"""ダウンロードしたCSVファイルを読み込む。"""
		last_path = super().attributes().csv_url().split('/')[-1]
		csv_file = super().attributes().base_directory() + os.sep + last_path
		csv_list = super().read_csv(csv_file)
		
		for index, key in enumerate(csv_list[0]):
			super().attributes().names()[index] = key

		for row in csv_list[1:]:
			tuple = Tuple(super().attributes(),row)
			print(tuple)
			super().table().add(tuple)
   

		(lambda x: x)(self) # NOP
